dotnet/src/Skills/Skills.OpenAPI/Extensions/RestApiOperationExtensions.cs

```diff
@@ -30,20 +30,26 @@ public static IReadOnlyList<RestApiOperationParameter> GetParameters(this RestAp
             RestApiOperationParameterStyle.Simple,
             defaultValue: operation.ServerUrl));
 
-        //Register the "input" parameter to be advertised and used for "text/plain" requests.
-        if (operation.Payload?.MediaType == MediaTypeTextPlain)
+        //Register the "payload" parameter to be advertised for Put and Post operations.
+        if (operation.Method == HttpMethod.Put || operation.Method == HttpMethod.Post)
```

Comment:
Looks like `OpenApiDocumentParserV30Tests.ItCanParsePutOperationMetadataSuccessfullyAsync` is failing.

Reply:
Yes, right, I did not run that unit tests subset locally :( Thanks, fixed.